packages/wallet/backend/src/walletAddress/service.ts

```diff
@@ -159,56 +124,38 @@ export class WalletAddressService implements IWalletAddressService {
         publicName: args.publicName,
         accountId: args.accountId,
         id: rafikiWalletAddress.id,
-        isWM: args.isWM,
-        assetCode: webMonetizationAsset?.code,
-        assetScale: webMonetizationAsset?.scale
+        assetCode: account.assetCode,
+        assetScale: this.env.BASE_ASSET_SCALE
       })
 
-      args.isWM &&
-        (await this.cache.set(walletAddress.id, walletAddress, {
-          expiry: 60
-        }))
+      await this.cache.set(walletAddress.id, walletAddress, {
+        expiry: 60
+      })
     }
 
     return walletAddress
   }
 
-  async list(userId: string, accountId: string): Promise<WalletAddressList> {
+  async list(userId: string, accountId: string): Promise<WalletAddress[]> {
     const account = await this.accountService.findAccountById(accountId, userId)
 
     const walletAddressesResult = await WalletAddress.query()
       .where('accountId', account.id)
       .where('active', true)
 
-    const result = walletAddressesResult.reduce(
-      (acc, pp) => {
-        if (pp.isWM) {
-          acc.wmWalletAddresses.push(pp)
-        } else {
-          acc.walletAddresses.push(pp)
-        }
-        return acc
-      },
-      {
-        wmWalletAddresses: [] as WalletAddress[],
-        walletAddresses: [] as WalletAddress[]
-      }
-    )
-
-    return result
+    return walletAddressesResult
   }
 
   async listAll(userId: string): Promise<WalletAddress[]> {
     return WalletAddress.query()
-      .where({ isWM: false, active: true })
+      .where({ active: true })
       .joinRelated('account')
       .where({
         'account.userId': userId
       })
   }
 
   async getById(args: GetWalletAddressArgs): Promise<WalletAddress> {
-    //* Cache only contains WalletAddresses with isWM = true
```

Comment:
as the method is performing additional validation on user and account, maybe caching should not be used checked in this case